connectors/sources/sharepoint_online.py

```diff
@@ -1223,21 +1302,33 @@ async def validate_config(self):
         if WILDCARD in configured_root_sites:
             return
 
-        remote_sites = []
+        retrieved_sites = []
 
         async for site_collection in self.client.site_collections():
             async for site in self.client.sites(
-                site_collection["siteCollection"]["hostname"], [WILDCARD]
+                site_collection["siteCollection"]["hostname"],
+                configured_root_sites,
+                self.configuration["enumerate_all_sites"],
             ):
-                remote_sites.append(site["name"])
+                if self.configuration["enumerate_all_sites"]:
+                    retrieved_sites.append(site["name"])
+                else:
+                    retrieved_sites.append(self._site_path_from_web_url(site["webUrl"]))
```

Comment:
Why do we need this? Won't the `get site by path` API return `name` field?

Reply:
No, not always. This is actually the root cause of some issues we've been seeing, where `/sites/outer/foo` and `/sites/foo` might both be named `foo`, and then you might have a `/sites/bar` named `Bar` (uppercase) and a `/sites/baz` named `my-baz`. There are no rules, and this is the wild west, aparantly. 💀